source/credentials_provider_sts_web_identity.c

```diff
@@ -0,0 +1,1112 @@
+        if (++sts_web_identity_user_data->attempt_count < STS_WEB_IDENTITY_MAX_ATTEMPTS &&
+            sts_web_identity_user_data->response.len) {
+            if (s_sts_web_identity_parse_retryale_error_from_response(
+                    sts_web_identity_user_data->allocator, &sts_web_identity_user_data->response)) {
+                s_sts_web_identity_query_credentials(sts_web_identity_user_data);
+                return;
+            }
+        }
+    }
+
+    s_sts_web_identity_finalize_get_credentials_query(sts_web_identity_user_data);
+}
+
+AWS_STATIC_STRING_FROM_LITERAL(s_sts_web_identity_accept_header, "Accept");
+AWS_STATIC_STRING_FROM_LITERAL(s_sts_web_identity_accept_header_value, "*/*");
+AWS_STATIC_STRING_FROM_LITERAL(s_sts_web_identity_user_agent_header, "User-Agent");
+AWS_STATIC_STRING_FROM_LITERAL(
+    s_sts_web_identity_user_agent_header_value,
+    "aws-sdk-crt/sts-web-identity-credentials-provider");
+AWS_STATIC_STRING_FROM_LITERAL(s_sts_web_identity_h1_0_keep_alive_header, "Connection");
+AWS_STATIC_STRING_FROM_LITERAL(s_sts_web_identity_h1_0_keep_alive_header_value, "keep-alive");
+
+static int s_make_sts_web_identity_http_query(
+    struct aws_credentials_provider_sts_web_identity_user_data *sts_web_identity_user_data,
+    struct aws_byte_cursor *uri) {
+    AWS_FATAL_ASSERT(sts_web_identity_user_data->connection);
+
+    struct aws_http_stream *stream = NULL;
+    struct aws_http_message *request = aws_http_message_new_request(sts_web_identity_user_data->allocator);
+    if (request == NULL) {
+        return AWS_OP_ERR;
+    }
+
+    struct aws_credentials_provider_sts_web_identity_impl *impl =
+        sts_web_identity_user_data->sts_web_identity_provider->impl;
+
+    struct aws_http_header accept_header = {
+        .name = aws_byte_cursor_from_string(s_sts_web_identity_accept_header),
+        .value = aws_byte_cursor_from_string(s_sts_web_identity_accept_header_value),
+    };
+    if (aws_http_message_add_header(request, accept_header)) {
+        goto on_error;
+    }
+
+    struct aws_http_header user_agent_header = {
+        .name = aws_byte_cursor_from_string(s_sts_web_identity_user_agent_header),
+        .value = aws_byte_cursor_from_string(s_sts_web_identity_user_agent_header_value),
+    };
+    if (aws_http_message_add_header(request, user_agent_header)) {
+        goto on_error;
+    }
+
+    struct aws_http_header keep_alive_header = {
+        .name = aws_byte_cursor_from_string(s_sts_web_identity_h1_0_keep_alive_header),
+        .value = aws_byte_cursor_from_string(s_sts_web_identity_h1_0_keep_alive_header_value),
+    };
+    if (aws_http_message_add_header(request, keep_alive_header)) {
+        goto on_error;
+    }
+
+    if (aws_http_message_set_request_path(request, *uri)) {
+        goto on_error;
+    }
+
+    if (aws_http_message_set_request_method(request, aws_byte_cursor_from_c_str("GET"))) {
```

Comment:
NIT: I think there's a predefined GET cursor somewhere in aws-c-http

Reply:
Yes, found it `aws_http_method_get`